patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch

```diff
@@ -73,26 +73,101 @@
- #if BUILDFLAG(FULL_SAFE_BROWSING)
- #include "chrome/browser/safe_browsing/advanced_protection_status_manager.h"
- #include "chrome/browser/safe_browsing/advanced_protection_status_manager_factory.h"
-+#endif
+@@ -24,7 +24,6 @@
+ #include "chrome/browser/password_manager/touch_to_fill_controller.h"
+ #include "chrome/browser/prerender/prerender_contents.h"
+ #include "chrome/browser/profiles/profile.h"
+-#include "chrome/browser/safe_browsing/chrome_password_protection_service.h"
+ #include "chrome/browser/signin/identity_manager_factory.h"
+ #include "chrome/browser/site_isolation/site_isolation_policy.h"
+ #include "chrome/browser/sync/profile_sync_service_factory.h"
+@@ -59,7 +58,6 @@
+ #include "components/password_manager/core/common/password_manager_features.h"
+ #include "components/password_manager/core/common/password_manager_pref_names.h"
+ #include "components/prefs/pref_service.h"
+-#include "components/safe_browsing/buildflags.h"
+ #include "components/sessions/content/content_record_password_state.h"
+ #include "components/signin/public/identity_manager/identity_manager.h"
+ #include "components/sync/driver/sync_service.h"
+@@ -87,12 +85,8 @@
+ #include "third_party/re2/src/re2/re2.h"
+ #include "url/url_constants.h"
+ 
+-#if BUILDFLAG(FULL_SAFE_BROWSING)
+-#include "chrome/browser/safe_browsing/advanced_protection_status_manager.h"
+-#include "chrome/browser/safe_browsing/advanced_protection_status_manager_factory.h"
  #include "ui/base/clipboard/clipboard.h"
  #include "ui/events/keycodes/keyboard_codes.h"
 -#endif
  
  #if defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
  #include "services/service_manager/public/cpp/connector.h"
-@@ -547,7 +547,7 @@ ChromePasswordManagerClient::GetPassword
+@@ -538,51 +532,15 @@ bool ChromePasswordManagerClient::IsIsol
+   return SiteIsolationPolicy::IsIsolationForPasswordSitesEnabled();
  }
- #endif
  
+-#if defined(ON_FOCUS_PING_ENABLED) || \
+-    defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
+-safe_browsing::PasswordProtectionService*
+-ChromePasswordManagerClient::GetPasswordProtectionService() const {
+-  return safe_browsing::ChromePasswordProtectionService::
+-      GetPasswordProtectionService(profile_);
+-}
+-#endif
+-
 -#if defined(ON_FOCUS_PING_ENABLED)
-+#if defined(ON_FOCUS_PING_ENABLED) && BUILDFLAG(FULL_SAFE_BROWSING)
- void ChromePasswordManagerClient::CheckSafeBrowsingReputation(
-     const GURL& form_action,
-     const GURL& frame_url) {
+-void ChromePasswordManagerClient::CheckSafeBrowsingReputation(
+-    const GURL& form_action,
+-    const GURL& frame_url) {
+-  safe_browsing::PasswordProtectionService* pps =
+-      GetPasswordProtectionService();
+-  if (pps) {
+-    pps->MaybeStartPasswordFieldOnFocusRequest(
+-        web_contents(), web_contents()->GetLastCommittedURL(), form_action,
+-        frame_url, pps->GetAccountInfo().hosted_domain);
+-  }
+-}
+-#endif  // defined(ON_FOCUS_PING_ENABLED)
+-
+ #if defined(SYNC_PASSWORD_REUSE_DETECTION_ENABLED)
```

Comment:
Do you know what sets `SYNC_PASSWORD_REUSE_DETECTION_ENABLED`? Is it a GN flag?

Reply:
`SYNC_PASSWORD_REUSE_DETECTION_ENABLED` is set by the following section of `components/password_manager/core/browser/BUILD.gn`:

```
# TODO(crbug.com/706392): Fix password reuse detection for Android.
password_reuse_detection_support = !is_android && !is_ios
password_on_focus_ping_support = !is_ios

config("password_reuse_detection_config") {
  defines = []
  if (password_reuse_detection_support) {
    defines += [ "SYNC_PASSWORD_REUSE_DETECTION_ENABLED" ]
  }
  if (password_on_focus_ping_support) {
    defines += [ "ON_FOCUS_PING_ENABLED" ]
  }
}
```

`ON_FOCUS_PING_ENABLED` is also set here as well.